cli/internal/context/context.go

```diff
@@ -455,8 +455,8 @@ func calculateGlobalHash(rootpath string, rootPackageJSON *fs.PackageJSON, pipel
 
 	if !util.IsYarn(packageManager.Name) {
 		// If we are not in Yarn, add the specfile and lockfile to global deps
-		globalDeps.Add(filepath.Join(rootpath, packageManager.Specfile))
-		globalDeps.Add(filepath.Join(rootpath, packageManager.Lockfile))
+		globalDeps.Add(filepath.Join(rootpath.ToStringDuringMigration(), packageManager.Specfile))
```

Comment:
I intentionally did not change this to `rootpath.Join(packageManager.Specfile)` because it is being added to an untyped set, and I wanted to address the types for that set separately.